test/test_settings.py

```diff
@@ -342,11 +342,22 @@ def test_cannot_set_invalid_values_for_max_header_list_size(self, val):
             assert e.value.error_code == h2.errors.ErrorCodes.PROTOCOL_ERROR
             assert s.max_header_list_size is None
 
-            with pytest.raises(h2.exceptions.InvalidSettingsValueError) as e:
+            with pytest.raises(h2.exceptions.InvalidSettingsValueError) as e: 
                 s[h2.settings.MAX_HEADER_LIST_SIZE] = val
 
             s.acknowledge()
             assert e.value.error_code == h2.errors.ErrorCodes.PROTOCOL_ERROR
 
             with pytest.raises(KeyError):
                 s[h2.settings.MAX_HEADER_LIST_SIZE]
+
+
+    def equal_objects_are_equal(s1,s2):
+        s1.acknowledge()
+        s2.acknowledge()
+        assert s1._settings==s2._settings
+   
+    def notequal_objects_are_notequal(s1,s2):
+        s1.acknowledge()
+        s2.acknowledge()
+        assert s1._settings!=s2._settings
```

Comment:
I have done most of the changes as you said but , tox is showing AssertionError with

"assert not s1 == s2"
or

"assert s1 != s2"
tox shows this error
assert <h2.settings.Settings object at 0x10a749410> != <h2.settings.Settings object at 0x10a749490>